### PR TITLE
Distill glue + Distillation

### PR DESCRIPTION
**Pull request by spacemanidol**

### CODE REVIEW

**Comment by spacemanidol** [comment]
No results so far. Running EXPs while cleaning up code to match standards

**Comment by spacemanidol** [comment]
pruning runs in progress. Results targeting 80,90,97 + distillation + 6 layer to follow. in the meantime can I get some reviews

**Comment by spacemanidol** [comment]
style fixes done.

File: integrations/transformers/run_distill_qa.py [hunk 1]
```diff
@@ -746,8 +750,8 @@ def prepare_validation_features(examples):
     # Start SparseML Integration
     ####################################################################################
     if training_args.do_train:
-        optim = load_optimizer(student_model, TrainingArguments)
-        steps_per_epoch = math.ceil(len(train_dataset) / (training_args.per_device_train_batch_size * training_args._n_gpu))
+        optim = load_optimizer(student_model, training_args)
+        steps_per_epoch = math.ceil(len(datasets["train"]) / (training_args.per_device_train_batch_size*training_args._n_gpu))
```
Comment (natuan) [inline integrations/transformers/run_distill_qa.py:754]: should use len(train_dataset)

File: integrations/transformers/run_distill_qa.py [hunk 2]
```diff
@@ -746,8 +750,8 @@ def prepare_validation_features(examples):
...
         manager = ScheduledModifierManager.from_yaml(data_args.nm_prune_config)
         training_args.num_train_epochs = float(manager.modifiers[0].end_epoch)
```
Comment (natuan) [inline integrations/transformers/run_distill_qa.py:756]: manager.max_epochs

**Review by natuan — CHANGES_REQUESTED** [review]

File: integrations/transformers/run_glue.py [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+        trainer.push_to_hub()
+
+    ####################################################################################
+    # Start SparseML Integration
+    ####################################################################################
+    if data_args.do_onnx_export:
+        logger.info("*** Export to ONNX ***")
+        os.environ["TOKENIZERS_PARALLELISM"] = "false"
+        exporter = ModuleExporter(
+            student_model, output_dir='onnx-export'
```
Comment (natuan) [inline integrations/transformers/run_glue.py:754]: output_dir=data_args.onnx_export_path

**Review by natuan — CHANGES_REQUESTED** [review]

File: integrations/transformers/README.md [hunk 4]
```diff
@@ -169,30 +173,57 @@ To explore the effect of model pruning compared to layer dropping, we train mode
...
+| bert-base-uncased     |97             |no       |no       |30          |18            |87.57         |83.34     |73.42          |
+| bert-base-uncased     |0              |yes      |no       |2           |0             |              |          |               |
+| bert-base-uncased     |80             |yes      |yes      |30          |18            |              |          |               |
+| bert-base-uncased     |90             |yes      |yes      |30          |18            |              |          |               |
+| bert-base-uncased     |97             |yes      |yes      |30          |18            |              |          |               |
 
-For any other projects combining Hugging Face transformers there are essentially four components to modify: imports and needed function, loading SparseML, modifying training script, and ONNX export. 
+## How to Integrate SparseML with Other Transformers Projects
+For any other projects using Hugging Face's transformers there are essentially four components to modify: imports and needed function, loading SparseML, modifying training script, and ONNX export.
```
[the remark this answers is not in the document]
Comment (spacemanidol) [reply integrations/transformers/README.md:226]: Cleaned up the wording. Main focus was on imports of libraries and functions required to load optimzier

**Review by bfineran — COMMENTED** [review]
looks good to me overall pending merge conflicts

File: integrations/huggingface-transformers/recipes/noprune10epoch.yaml [hunk 5]
```diff
@@ -2,5 +2,5 @@ version: 1.1.0
 
 modifiers:
     - !EpochRangeModifier
-        end_epoch: 10.0
+        end_epoch: 30.0
```
Comment (natuan) [inline integrations/huggingface-transformers/recipes/noprune10epoch.yaml:5]: should this file now be changed to "noprune30epochs.yaml"?

Comment (spacemanidol) [reply integrations/huggingface-transformers/recipes/noprune10epoch.yaml:5]: Done

**Review by natuan — APPROVED** [review]

**Review by bfineran — APPROVED** [review]
great stuff @spacemanidol!

**Review by markurtz — APPROVED** [review]

**Pull request merged**